hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -227,12 +232,23 @@ public KafkaOffsetGen(TypedProperties props) {
   // else return earliest offsets
   private Map<TopicPartition, Long> checkupValidOffsets(KafkaConsumer consumer,
                                                         Option<String> lastCheckpointStr, Set<TopicPartition> topicPartitions) {
-    Map<TopicPartition, Long> checkpointOffsets = CheckpointUtils.strToOffsets(lastCheckpointStr.get());
     Map<TopicPartition, Long> earliestOffsets = consumer.beginningOffsets(topicPartitions);
+    if (checkTopicCheckPoint(lastCheckpointStr)) {
+      Map<TopicPartition, Long> checkpointOffsets = CheckpointUtils.strToOffsets(lastCheckpointStr.get());
+      boolean checkpointOffsetReseter = checkpointOffsets.entrySet().stream()
+              .anyMatch(offset -> offset.getValue() < earliestOffsets.get(offset.getKey()));
+      return checkpointOffsetReseter ? earliestOffsets : checkpointOffsets;
+    }
+
+    switch (autoResetValue) {
+      case EARLIEST:
+        return earliestOffsets;
+      case LATEST:
+        return consumer.endOffsets(topicPartitions);
+      default:
```

Comment:
is it intentional for NONE we will throw exception? also, may I know where is NONE(KafkaResetOffsetStrategies) used?

Reply:
I also don’t think it’s necessary to add NONE